app/code/Magento/Eav/Model/ResourceModel/ReadHandler.php

```diff
@@ -91,6 +93,37 @@ protected function getContextVariables(ScopeInterface $scope)
      */
     public function execute($entityType, $entityData, $arguments = [])
     {
+        return $this->getData($entityType, $entityData, $arguments = []);
+    }
+
+    /**
+     * @param string $entityType
+     * @param array $entityData
+     * @param array $arguments
+     * @return array
+     * @throws \Exception
+     * @throws \Magento\Framework\Exception\ConfigurationMismatchException
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function getAllScopeData($entityType, $entityData, $arguments = [])
+    {
+        return $this->getData($entityType, $entityData, $arguments = [], true);
+    }
+
+    /**
+     * @param string $entityType
+     * @param array $entityData
+     * @param array $arguments
+     * @param bool $loadAllScopes
+     * @return array
+     * @throws \Exception
+     * @throws \Magento\Framework\Exception\ConfigurationMismatchException
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function getData($entityType, $entityData, $arguments = [], $loadAllScopes = false)
```

Comment:
We moved the code from the `execute` method into this `getData` method so that we could make it work for both `getAllScopeData` and `execute`.